pkg/recipes/terraform/install.go

```diff
@@ -78,5 +81,40 @@ func Install(ctx context.Context, installer *install.Installer, tfDir string) (s
 
 	logger.Info(fmt.Sprintf("Terraform latest version installed to: %q", execPath))
 
-	return execPath, nil
+	// Create a new instance of tfexec.Terraform with current Terraform installation path
+	tf, err := NewTerraform(ctx, tfDir, execPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Verify Terraform installation is complete before proceeding
+	for attempt := 0; attempt <= installRetryCount; attempt++ {
```

Comment:
It seems like the issue is that the Terraform installation doesn't complete fully before we try to use the executable. The idea is to wrap checking the TF version with retries so we only continue when we know for sure that calling the executable won't cause issues

Reply:
If `installRetryCount` is 5 then this will run 6 times. Why not make it just less than so that we can also get rid of the `if case` in the loop?

Reply:
Don't think we can completely get rid of the if case since that's how we decide whether to retry or not. The retry count is 5, not necessarily the number of times it runs so after the 5th retry, we'd return the error if it's still happening. We can change this number though, 5 just seemed like a logical starting point. Thoughts on the right value for that? @kachawla

Reply:
Yeah we don't really have a good way to decide on the right retry count until we have some metrics on how long it takes for installation to complete.

@sk593, I think @ytimocin wasn't asking the reasoning for retry count value, but instead that this loop will be executed retryCount+1 times since it starts at 0 `for attempt := 0; attempt <= installRetryCount; attempt++ {`

Reply:
My bad, I misunderstood the question. I updated the attempts to start at 1 so we get the proper count

Reply:
Yes, @kachawla thanks for the clarification! We can also start from 0 and go to `upperLimit - 1` @sk593.

Reply:
Going to use the suggested so we get the proper number of logs